aws-xray-propagator/src/main/java/io/opentelemetry/contrib/awsxray/propagator/AwsXrayPropagator.java

```diff
@@ -112,49 +108,20 @@ public <C> void inject(Context context, @Nullable C carrier, TextMapSetter<C> se
     char samplingFlag = spanContext.isSampled() ? IS_SAMPLED : NOT_SAMPLED;
     // TODO: Add OT trace state to the X-Ray trace header
 
-    StringBuilder traceHeader = new StringBuilder();
-    traceHeader
-        .append(TRACE_ID_KEY)
-        .append(KV_DELIMITER)
-        .append(xrayTraceId)
-        .append(TRACE_HEADER_DELIMITER)
-        .append(PARENT_ID_KEY)
-        .append(KV_DELIMITER)
-        .append(parentId)
-        .append(TRACE_HEADER_DELIMITER)
-        .append(SAMPLED_FLAG_KEY)
-        .append(KV_DELIMITER)
-        .append(samplingFlag);
-
-    Baggage baggage = Baggage.fromContext(context);
-    // Truncate baggage to 256 chars per X-Ray spec.
-    baggage.forEach(
-        new BiConsumer<String, BaggageEntry>() {
-
-          private int baggageWrittenBytes;
-
-          @Override
-          public void accept(String key, BaggageEntry entry) {
-            if (key.equals(TRACE_ID_KEY)
-                || key.equals(PARENT_ID_KEY)
-                || key.equals(SAMPLED_FLAG_KEY)) {
-              return;
-            }
-            // Size is key/value pair, excludes delimiter.
-            int size = key.length() + entry.getValue().length() + 1;
-            if (baggageWrittenBytes + size > 256) {
```

Comment:
> The primary incompatibility is that the limit of 256 characters in the Amazon Trace header. This is incompatible with the [W3C baggage limit](https://www.w3.org/TR/baggage/#limits) of 8192 bytes. Attempting to truncate the baggage to fit within this limit risks corrupting the entries and should not be attempted.

It looks like the code is only trying to include baggage entires if the entire entry can be included without exceeding 256 bytes.

When you talk about corrupting entries, are you referring to some entries being present while others not? Can you help me understand why this is problematic?